src/impl/scanner.ts

```diff
@@ -17,6 +17,10 @@ export function createScanner(text: string, ignoreTrivia: boolean = false): JSON
 		value: string = '',
 		tokenOffset = 0,
 		token: SyntaxKind = SyntaxKind.Unknown,
+		lineNumber = 0,
+		tokenLineNumber = 0,
+		lineEndOffset = 0,
```

Comment:
I'd call it lineStartOffset (the start offset of the current line) and tokenLineStartOffset (the start offset of the token line)